src/streaming-client.ts

```diff
@@ -0,0 +1,156 @@
+import * as fs from 'fs';
+import * as sockets from 'websocket';
+import * as events from 'events';
+import { Duplex, PassThrough } from 'stream';
+
+import { AudioConfig } from './models/streaming/AudioConfig';
+import { BufferedDuplex } from './models/streaming/BufferedDuplex';
+import { 
+    StreamingResponse, 
+    StreamingHypothesis, 
+    StreamingConnected 
+} from './models/streaming/StreamingResponses';
+
+/**
+ * Client which handles a streaming connection to the Rev.ai API.
+ *
+ * @param accessToken {string} Access token associated with the user's account
+ * @param config {AudioConfig} Configuration of the audio the user will send from this client
+ * @param version (optional) {string} Version of the Rev.ai API the user wants to use
```

Comment:
What is the industry convention for where to put docs for constructor parameters? Do we put them by the constructor definition?

Reply:
It actually looks like google documents these on the constructor method itself. I'll copy that convention